image_transport/src/camera_subscriber.cpp

```diff
@@ -82,22 +93,25 @@ struct CameraSubscriber::Impl
   {
     int threshold = 3 * both_received_;
     if (image_received_ > threshold || info_received_ > threshold) {
+      std::string info_topic;
+      info_topic = info_sub_.getTopic();
       RCLCPP_WARN(
         logger_,
         "[image_transport] Topics '%s' and '%s' do not appear to be synchronized. "
         "In the last 10s:\n"
         "\tImage messages received:      %d\n"
         "\tCameraInfo messages received: %d\n"
         "\tSynchronized pairs:           %d",
-        image_sub_.getTopic().c_str(), info_sub_.getTopic().c_str(),
+        image_sub_.getTopic().c_str(), info_topic.c_str(),
         image_received_, info_received_, both_received_);
     }
     image_received_ = info_received_ = both_received_ = 0;
   }
 
+  std::shared_ptr<NodeType> node_;
```

Comment:
by holding a reference to the Node here, are we going to end up with some sort of hanging reference where the node never goes out of scope?